flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/SqlGatewayServiceStatementITCase.java

```diff
@@ -0,0 +1,294 @@
+        for (TestSqlStatement statement : statements) {
+            StringBuilder builder = new StringBuilder();
+            builder.append(statement.getComment());
+            builder.append(statement.getSql());
+
+            try {
+                builder.append(runSingleStatement(sessionHandle, statement.getSql()));
+            } catch (Throwable t) {
+                Throwable root = getRootCause(t);
+                builder.append(
+                        Tag.ERROR.addTag(
+                                root.getClass().getName()
+                                        + ": "
+                                        + root.getMessage().trim()
+                                        + "\n"));
+            }
+            output.add(builder.toString());
+        }
+
+        return output;
+    }
+
+    // -------------------------------------------------------------------------------------------
+    // Utility
+    // -------------------------------------------------------------------------------------------
+
+    /** Mark the output type. */
+    enum Tag {
+        INFO("!info"),
+
+        OK("!ok"),
+
+        ERROR("!error");
+
+        private final String tag;
+
+        Tag(String tag) {
+            this.tag = tag;
+        }
+
+        public String addTag(String content) {
+            return HINT_START_OF_OUTPUT + "\n" + content + tag + "\n";
+        }
+    }
+
+    private String getInputFromPath(String sqlPath) throws IOException {
+        URL url = SqlGatewayServiceStatementITCase.class.getResource("/" + sqlPath);
+
+        // replace the placeholder with specified value if exists
+        String[] keys = replaceVars.keySet().toArray(new String[0]);
+        String[] values = Arrays.stream(keys).map(replaceVars::get).toArray(String[]::new);
+
+        return StringUtils.replaceEach(
+                IOUtils.toString(checkNotNull(url), StandardCharsets.UTF_8), keys, values);
+    }
+
+    protected static Throwable getRootCause(Throwable e) {
+        Throwable root = e;
+        while (root.getCause() != null) {
+            root = root.getCause();
+        }
+        return root;
+    }
+
+    private String runSingleStatement(SessionHandle sessionHandle, String statement)
+            throws Exception {
+        OperationHandle operationHandle =
+                service.executeStatement(sessionHandle, statement, -1, new Configuration());
+        CommonTestUtils.waitUtil(
+                () ->
+                        service.getOperationInfo(sessionHandle, operationHandle)
+                                .getStatus()
+                                .isTerminalStatus(),
+                Duration.ofSeconds(100),
+                "Failed to wait operation finish.");
+
+        if (!service.getOperationInfo(sessionHandle, operationHandle).isHasResults()) {
+            return Tag.INFO.addTag("");
+        }
+
+        if (statement.toLowerCase().startsWith("explain")
```

Comment:
Why them need to process specially? The logic of the judgment is a bit hack? Does exist other special case
Add some comments would be better.

Reply:
> Why them need to process specially?

The current logic aligns with the SQL Client. The reason here is the result of the `explain` and `show create` in the table is pretty. It's more straightforward to print the content of the result. 

>  The logic of the judgment is a bit hack?

The current implementation just matches the keyword. In the future(in the next version), I will introduce a `TokenManager` that is able to recognize all the keywords in Flink SQL. Considering this is just a test here and logic is controllable, I think it's acceptable to do this.